.github/scripts/build_publish_nightly_docker.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -xeuo pipefail
+
+git fetch origin nightly
```

Comment:
We might not have to do it on the `nightly` branch in particular, with github actions we should be able to get away with just running it directly on master

Reply:
How do we get the nightly's SHA without fetching the nightly branch? Since I am tagging the docker with the nightly's commit. For example, the SHA for 2021-02-04 nightly release is "1518aee".

Reply:
Right, in the future we won't be using the `nightly` branch anymore since GHA can run a scheduled workflow with credentials without having to be on a separate branch.

If you do want to use the nightly branch though I guess we could. Another possibility is that the `sha` is actually baked into the pytorch binary itself and you can access it with 
```
python -c 'import torch; print(torch.version.git_version)'
```

Reply:
I see, I removed the nightly branch dependency by using "docker run" to extract git version of pytorch binary.